src/markdown.cpp

```diff
@@ -1856,15 +1899,33 @@ static QCString extractTitleId(QCString &title, int level)
   {
     std::string id = match[1].str();
     title = title.left(match.position());
+    int &count = id_count[id];
+    if (count > 0) {
+      warn(m_fileName,
+           m_lineNr,
+           "A automatically generated id already has the name '%s'!",
+           id.c_str());
+    }
     //printf("found match id='%s' title=%s\n",id.c_str(),qPrint(title));
     AUTO_TRACE_EXIT("id={}",id);
-    return QCString(id);
+    return id;
   }
-  if ((level > 0) && (level <= Config_getInt(TOC_INCLUDE_HEADINGS)))
-  {
-    static AtomicInt autoId { 0 };
+  if ((level > 0) && (level <= Config_getInt(TOC_INCLUDE_HEADINGS))) {
+    static HEADING_AUTO_IDENTIFIER_t behaviour = Config_getEnum(HEADING_AUTO_IDENTIFIER);
```

Comment:
'behaviour' -> 'behavior'
(UK-English versus US-English).
Also in other files.